index.html

```diff
@@ -4975,6 +4975,634 @@ <h2 id="AudioWorkerParamDescriptor">
           </section>
         </section>
       </section>
+      <section>
+        <h2 id="AudioWorklet">
+          The <dfn>AudioWorklet</dfn> interface
+        </h2>
+        <p>
+          The <code>AudioWorklet</code> object allows importing user-supplied
+          Javascript code that runs on the audio <a>rendering thread</a>. This
+          processing mechanism ensures the synchronous execution of the script
+          code with other built-in <code>AudioNode</code>s in the audio graph.
+        </p>
+        <p>
+          The <code>AudioWorklet</code> object stores the definitions of a
+          custom audio processor and an <code>AudioWorkletNode</code> object
+          can be constructed based on the processor definition. Note that the
+          registered definition in an AudioWorklet is valid for the entire
+          global scope (i.e. <code>window</code>), therefore any
+          <a><code>AudioContext</code></a> instance in the same global scope
+          can access the registered definition to create a custom node.
+        </p>
+        <p>
+          After importing a module script into the
+          <a><code>AudioWorkletGlobalScope</code></a>, an instance of
+          <a><code>AudioWorkletNode</code></a> can be created in the main
+          global scope. At the same time, the user agent automatically creates
+          an instance of <a><code>AudioWorkletProcessor</code></a>. This
+          counterpart represents an audio processing module that runs on the
+          audio <a>rendering thread</a>. These two components can communicate
+          via message passing.
+        </p>
+        <dl title="interface AudioWorklet : Worklet" class="idl">
+          <dt>
+            Promise&lt;void&gt; import(DOMString moduleUrl)
+          </dt>
+          <dd>
+            <p>
+              Imports a module script at <code>moduleURL</code>. The returned
+              promise is resolved when the code is successfully imported and
+              parsed. It is rejected when the code contains a syntax error, the
+              file is not found at the specified location, or there is a name
+              conflict between registered
+              <a><code>AudioWorkletProcessor</code></a> definitions.
+            </p>
```

Comment:
I think we need to specify in more detail how a name conflict results in the failure of this Promise. For example, does a single `registerAudioWorkletProcessor` call failure invalidate the effect of other successful calls? Is the failure the result of an exception being thrown in the AWGS by a failed registration during imported script? If so, what if the user were to wrap the failed registration in a try/catch: would the Promise still fail? Or is it the result of some magic failure state being recorded by the UA and propagated out to the Promise? 

I believe it is simplest to specify that the Promise fails if any uncaught exception occurs during the execution of the imported script, and to clarify that `registerAudioWorkletProcessor` throws an exception if a name conflict occurs.